lib/adapters/QUnitAdapter.js

```diff
@@ -58,27 +88,23 @@ export default class QUnitAdapter extends EventEmitter {
   }
 
   onTestDone (details) {
-    var test = this.idDataMap[details.testId]
-
-    if (details.failed !== 0) {
-      test.status = 'failed'
-    } else if (details.skipped === true) {
-      test.status = 'skipped'
-    } else {
-      test.status = 'passed'
-    }
+    var test
 
-    test.runtime = details.runtime
-    test.errors = this.errors
+    details.errors = this.errors
+    test = this.convertTest(details)
 
     this.emit('testEnd', test)
   }
 
   onModuleDone (details) {
-    this.emit('suiteEnd', this.idDataMap[details.name])
+    if (details.name !== '') {
```

Comment:
This isn't a great check, and its now scattered across multiple places in this class. Let's try to at least abstract it, so that we can call `if (this.globalModule(details)) { [...]`.